staff/web/vhost/vhosting_check.py

```diff
@@ -67,27 +101,40 @@ def check_vhosting():
         for vhost_url in vhosts.keys():
             if any(is_special(url) for url in {vhost_url} | set(vhosts[vhost_url]['aliases'])):
                 continue
-            vhost_urls.append('http://' + vhost_url + '\n')
+            vhost_urls.append('http://' + vhost_url)
         # For log niceness
         vhost_urls.sort()
         for site in vhost_urls:
             try:
-                site_opened = req.urlopen(site, timeout=10)
-                site_html = site_opened.readall().decode('utf-8')
-                counter = False
+                print("Opening", site)
+                site_html = r.get(site, timeout=10).text
+
+                isMissingDisc = False
+                if unavailable_site.search(site_html):
+                    raise Exception("Hosting down")
                 if not disclaimer_pattern.search(site_html):
-                    counter = True
-                    m_d.writelines(site)
+                    isMissingDisc = True
+                    m_d.writelines(site + "\n")
+                    print("\tNo disclaimer")
                 # Check if they have any OCF banner in their images
                 img_urls = {img_url if img_url.startswith('http') else site + '/' + img_url for img_url in
                             img_regex.findall(site_html)}
-                img_hashes = {hashlib.md5(requests.get(img_url).content).hexdigest() for img_url in img_urls}
-                if not banner_hashes & img_hashes:
-                    m_i.writelines(site)
-                    if counter:
-                        m_b.writelines(site)
+                for img_url in img_urls:
+                    #I'm not exactly sure what the deal with \r is but this should work
+                    idata = r.get(img_url)
+                    img_hash_raw = hashlib.md5(idata.content).hexdigest()
+                    img_hash_unix = hashlib.md5(idata.content.replace(b'\r\n', b'\n')).hexdigest()
+                    img_hash_window = hashlib.md5(idata.content.replace(b'\n', b'\r\n')).hexdigest()
+                    if any([x in banner_hashes for x in [img_hash_raw, img_hash_unix, img_hash_window]]):
+                        break
```

Comment:
this is... bizarre. i assume this is for the SVGs?

Reply:
I found this issue when looking at PNGs. Certain versions had carriage returns in them and others didn't. This "workaround" should permit hash verification however the carriage returns are setup. (It may apply to SVGs too, I'm not sure)